nemo/collections/tts/models/fastpitch.py

```diff
@@ -515,22 +516,26 @@ def list_available_models(cls) -> 'List[PretrainedModelInfo]':
     def _prepare_for_export(self, **kwargs):
         super()._prepare_for_export(**kwargs)
 
+        tensor_shape = ('T') if self.export_config["enable_ragged_batches"] else ('B', 'T')
+
         # Define input_types and output_types as required by export()
         self._input_types = {
-            "text": NeuralType(('B', 'T_text'), TokenIndex()),
-            "pitch": NeuralType(('B', 'T_text'), RegressionValuesType()),
-            "pace": NeuralType(('B', 'T_text'), optional=True),
-            "volume": NeuralType(('B', 'T_text')),
-            "speaker": NeuralType(('B'), Index()),
+            "text": NeuralType(tensor_shape, TokenIndex()),
+            "pitch": NeuralType(tensor_shape, RegressionValuesType()),
+            "pace": NeuralType(tensor_shape),
+            "speaker": NeuralType(('B'), Index(), optional=True),
+            "volume": NeuralType(tensor_shape, optional=True),
+            "batch_lengths": NeuralType(('B'), optional=True),
         }
         self._output_types = {
-            "spect": NeuralType(('B', 'D', 'T_spec'), MelSpectrogramType()),
+            "spect": NeuralType(('B', 'D', 'T'), MelSpectrogramType()),
             "num_frames": NeuralType(('B'), TokenDurationType()),
-            "durs_predicted": NeuralType(('B', 'T_text'), TokenDurationType()),
-            "log_durs_predicted": NeuralType(('B', 'T_text'), TokenLogDurationType()),
-            "pitch_predicted": NeuralType(('B', 'T_text'), RegressionValuesType()),
-            "volume_aligned": NeuralType(('B', 'T_spec'), RegressionValuesType()),
+            "durs_predicted": NeuralType(('B', 'T'), TokenDurationType()),
+            "log_durs_predicted": NeuralType(('B', 'T'), TokenLogDurationType()),
+            "pitch_predicted": NeuralType(('B', 'T'), RegressionValuesType()),
```

Comment:
Why remove the distinction between `T_text` and `T_spec`?

Reply:
Because export was crashing with some message like can't do NeuralType of "_". At inference, NeuralTypes don't exist anyways so I didn't root cause it